### ISSUE TITLE
Train text encoder in train_text_to_image_sdxl.py

### ISSUE THREAD

**Issue by stpg06** [huggingface/diffusers#7536]
I have used this script successfully on a A100 40GB to train a full SDXL model, I would like to try training the text_encoder as well and that is currently not an option. 

I'm wondering if that option can be added with possibly a separate learning rate argument for it.  I know things are done to make things simple and basic for inexperienced users like myself and I am highly appreciative of all the hard work you guys put in here. I just see no reason why this option is confined to the LoRa scripts but training the text_encoder isn't at least optional in the full script. I think it's been well documented that training the text encoder helps in all cases. It may crash the GPU, still be nice to see if I could get it to run by tweaking other settings.

Again, thanks for all this team does, and if there is some other script that trains both the full model and text_encoder, please let me know.

**Comment by bghira** [comment]
tuning a low-rank adaptation network applies to parts of the text encoder, and is much easier to train and apply at different strength levels.

this requires pretty extreme quantity of vram and balloons the complexity of the scripts. the proper way to tune the text encoder is using the OpenCLIP codebase from ml_foundations. this requires a **lot** of training data - in the hundreds of thousands at a starting point. contrastive loss requires a lot of compute.

**Comment by stpg06** [comment]
> tuning a low-rank adaptation network applies to parts of the text encoder, and is much easier to train and apply at different strength levels.
> 
> this requires pretty extreme quantity of vram and balloons the complexity of the scripts. the proper way to tune the text encoder is using the OpenCLIP codebase from ml_foundations. this requires a **lot** of training data - in the hundreds of thousands at a starting point. contrastive loss requires a lot of compute.

I see, so it's not just as simple as the Dreambooth method like with SD1-2? In those scripts you can train the text encoder easily. I know it's a larger model, but at the same time the text encoders are much  smaller than the unet model, so I wouldn't expect it to add that much to the process.  You'd  think it might still stick just under 40GB on an A100. Perhaps I'm just not understanding the difference between Dreambooth scripts and the train_text_to_image scripts. Also, since the capability to train SDXL  text encoders when training LoRas, I guess I'm just confused as to why we can't have full model training with that capability. I'm not talking about training a whole text encoder, only fine-tuning the SDXL ones, just so it's clear.

**Comment by bghira** [comment]
trained weights have to be stored in fp32, which increases the vram consumption a lot more over inference time

**Comment by stpg06** [comment]
Wow, that is high. It's too bad you can't do it in separate fine-tuning sessions. I guess then the text encoder wouldn't match the unet model, though. I accept that is too hard  and memory consuming based on what you've said. 

My issue with the LoRa training is that it inherits too much of the style from my images, whereas I prefer training on a subject, so the LoRa's often come out useless to me, too much memory on the style and faces, etc, not enough with the subject or complexities. SDXL seems to have adequate parameters to train on more complex subjects, but without the ability to train the text encoder to bake in fine and important details, it usually makes a mess of anything I can do with it. I guess it's just that the bigger and better these models get, the harder it is to work with them.  Soon we won't be able to run inference with them at all, big companies will control them and charge you a fortune.

**Comment by sayakpaul** [comment]
Closing because of inactivity.
